packages/client/lib/AccessTokenClientV1_0_11.ts

```diff
@@ -91,10 +96,18 @@ export class AccessTokenClientV1_0_11 {
           : undefined,
     });
 
-    return this.sendAuthCode(requestTokenURL, accessTokenRequest);
+    let dPoP: string | undefined;
+    if (createDPoPOptions?.dPoPSigningAlgValuesSupported && createDPoPOptions.dPoPSigningAlgValuesSupported.length > 0) {
+      const htu = requestTokenURL.split('?')[0].split('#')[0];
```

Comment:
This is being implemented clientside and only takes the tokenUrl from the metadata, so I think the URL should be fine. What checks would you implement?

Reply:
The split first strips of the query and then strips of the fragment. I am not sure what the preconditions are of the requestTokenURL, but I would expect a bit of expressions checking whether your preconditions is actually satisfied.

Anyway since you are doing the same splits somewhere else I would create a function at least.